main.go

```diff
@@ -0,0 +1,49 @@
+package main
+
+import (
+	"./authFunctions"
+
+	"database/sql"
+	"log"
+	"net/http"
+
+	_ "github.com/go-sql-driver/mysql"
+	"github.com/gorilla/mux"
+)
+
+func signup(w http.ResponseWriter, r *http.Request) {
+	authFunctions.Signup(w, r)
+}
+
+func login(w http.ResponseWriter, r *http.Request) {
+	authFunctions.Login(w, r)
+}
+
+var db *sql.DB
+
+func main() {
+	db, err := sql.Open("mysql", authFunctions.DBLocation())
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	err = db.Ping()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	defer db.Close()

```

Comment:
プログラム起動時にDB接続を確認してるかんじかな？
接続確認したら、すぐにコネクションクローズしても良いと思うよ。
```suggestion
	db, err := sql.Open("mysql", authFunctions.DBLocation())
	if err != nil {
		log.Fatal(err)
	}

	err = db.Ping()
	if err != nil {
		log.Fatal(err)
	}

	db.Close()

```

Reply:
わかりました，ありがとうございます！